simpleclient_hotspot/src/main/java/io/prometheus/client/hotspot/StandardExports.java

```diff
@@ -59,33 +56,34 @@ public StandardExports() {
     List<MetricFamilySamples> mfs = new ArrayList<MetricFamilySamples>();
 
     try {
-      /*
-      There are two interfaces com.ibm.lang.management.OperatingSystemMXBean and com.sun.management.OperatingSystemMXBean,
-      but no common interface which exposes getProcessCpuTime. Hence call on the implementing class, which is default access,
-      so use reflection hack to set it accessible.
-      */
-      Method method = osBean.getClass().getMethod("getProcessCpuTime", null);
-      if (!method.isAccessible()) {
-        method.setAccessible(true);
-      }
-      Long l = (Long) method.invoke(osBean);
+      // There exist at least 2 UnixOperatingSystemMXBean interfaces, in com.sun.management and 
+      // com.ibm.lang.management. There are also environments (such as Wildfly) where access to these
+      // interfaces is restricted. Hence use reflection and recursively go through implemented
+      // interfaces until the method can be made accessible and invoked.
+      Long processCpuTime = callLongGetter("getProcessCpuTime", osBean);
       mfs.add(new CounterMetricFamily("process_cpu_seconds_total", "Total user and system CPU time spent in seconds.",
-          l / NANOSECONDS_PER_SECOND));
+          processCpuTime / NANOSECONDS_PER_SECOND));
     }
     catch (Exception e) {
       LOGGER.log(Level.FINE,"Could not access process cpu time", e);
     }
 
-
     mfs.add(new GaugeMetricFamily("process_start_time_seconds", "Start time of the process since unix epoch in seconds.",
         runtimeBean.getStartTime() / MILLISECONDS_PER_SECOND));
 
-    if (unix) {
-      UnixOperatingSystemMXBean unixBean = (UnixOperatingSystemMXBean) osBean;
-      mfs.add(new GaugeMetricFamily("process_open_fds", "Number of open file descriptors.",
-          unixBean.getOpenFileDescriptorCount()));
-      mfs.add(new GaugeMetricFamily("process_max_fds", "Maximum number of open file descriptors.",
-          unixBean.getMaxFileDescriptorCount()));
+    // There exist at least 2 UnixOperatingSystemMXBean interfaces, in com.sun.management and 
+    // com.ibm.lang.management. There are also environments (such as Wildfly) where access to these
+    // interfaces is restricted. Hence use reflection and recursively go through implemented
```

Comment:
I don't think restricted is quite the right word, my understanding is that the way Wildfly does things breaks standard JMX usage.